components/helm-broker/internal/bundle/syncer.go

```diff
@@ -121,15 +153,80 @@ func (s *Syncer) removeAndNotifyAboutDuplication(allBundles []CompleteBundle) []
 
 		var msgs []string
 		for _, duplicated := range bundles {
-			msgs = append(msgs, fmt.Sprintf("[url: %s, name: %s, version: %s]", duplicated.Bundle.RemoteRepositoryURL, duplicated.Bundle.Name, duplicated.Bundle.Version.String()))
+			msgs = append(msgs, fmt.Sprintf("[url: %s, name: %s, version: %s]", duplicated.Bundle.RepositoryURL, duplicated.Bundle.Name, duplicated.Bundle.Version.String()))
 		}
 		s.log.Warnf("There are more than one bundle with the same ID (%s): %s", id, strings.Join(msgs, ", "))
+
+		if err := s.ensureDocsTopicsRemoved(string(id)); err != nil {
+			return nil, errors.Wrapf(err, "while ensuring ClusterDocsTopic %s is removed", id)
+		}
+	}
+
+	return uniqueBundles, nil
+}
+
+func (s *Syncer) ensureDocsTopics(bundle CompleteBundle) error {
+	if err := s.ensureDocsTopicsRemoved(string(bundle.ID())); err != nil {
```

Comment:
you should not remove the CDT. You should just create a CDT if `IsAlreadyExist` error is returned then execute Update.  Deleting and Creating is a different flow that Create or Updating. The CMS controller will work in totally different manner